Add examples for `Dual` (#239)
typelevel/monoids: docs/src/main/mdoc/index.md

Last(1.some) |+| Last(Option.empty[Int]) |+| Last(Option.empty[Int])
```

> As mentioned in the beginning `Dual` inverts the `combine` operation of a `monoid`.

Reviewer: ```suggestion
As mentioned above, `Dual` inverts the `combine` operation of a `Monoid`.
```

> However, this can only be clearly observed when the algebra is not commutative.

Reviewer: ```suggestion
That could be well observed for the non-commutative algebra.
```
